x-pack/plugins/lens/public/indexpattern_datasource/field_item.tsx

```diff
@@ -347,6 +349,41 @@ function FieldItemPopoverContents(props: FieldItemProps) {
     />
   );
 
+  const exploreInDiscover = useMemo(() => {
+    const meta = {
+      id: indexPattern.id,
+      columns: [field.name],
+      filters: {
+        enabled: {
+          lucene: [],
+          kuery: [],
+        },
+        disabled: {
+          lucene: [],
+          kuery: [],
+        },
+      },
+    };
+    const { filters: newFilters, query: newQuery } = combineQueryAndFilters(
+      query,
+      filters,
+      meta,
+      [indexPattern],
+      getEsQueryConfig(services.uiSettings)
+    );
+
+    if (!services.discover) {
```

Comment:
Hi! Just noticed that we should also check for `core.application.capabilities.discover.show` and hide the button in this case.

Reply:
Thanx @jughosta, you are right :) I created a followup PR here https://github.com/elastic/kibana/pull/142332